package/MDAnalysis/analysis/base.py

```diff
@@ -85,21 +88,170 @@ def _prepare(self):
 
     def _conclude(self):
         """Finalise the results you've gathered.
-
         Called at the end of the run() method to finish everything up.
         """
         pass
 
-    def run(self):
+    def run(self, parallel=None, threads=None):
         """Perform the calculation"""
         logger.info("Starting preparation")
-        self._prepare()
-        for i, ts in enumerate(
-                self._trajectory[self.start:self.stop:self.step]):
-            self._ts = ts
-            #logger.info("--> Doing frame {} of {}".format(i+1, self.nframes))
-            self._single_frame()
-        logger.info("Finishing up")
+        if parallel is None:
+            self._prepare()
+            for i, ts in enumerate(
+                    self._trajectory[self.start:self.stop:self.step]):
+                #logger.info("--> Doing frame {} of {}".format(i+1, self.nframes))
+                self._single_frame(ts)
+            #logger.info("Finishing up")
+            self._conclude()
+
+        else:
+            if threads is None:
```

Comment:
I'd prefer if all this code was contained inside `_parallel_run`, the run function can just end up as:
``` python
def run(self, parallel=False, nthreads=None):
    if not parallel:
        self._serial_run()
    else:
        self._parallel_run(nthreads)
```

So `_serial_run` just contains the "original" run code, and parallel run has your new code